src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1198,7 +1198,16 @@ static void ValidateLength(string decoratorName, DecoratorSyntax decoratorSyntax
                     .WithDescription("Marks an object parameter as only permitting properties specifically included in the type definition")
                     .WithFlags(FunctionFlags.ParameterOrTypeDecorator)
                     .WithAttachableType(LanguageConstants.Object)
-                    .WithValidator(ValidateNotTargetingAlias)
+                    .WithValidator((decoratorName, decoratorSyntax, targetType, typeManager, binder, diagnosticWriter) =>
+                    {
+                        ValidateNotTargetingAlias(decoratorName, decoratorSyntax, targetType, typeManager, binder, diagnosticWriter);
+
+                        // make sure the target type doesn't have an explicit additional properties declaration
+                        if ((targetType as ObjectType)?.AdditionalPropertiesFlags.HasFlag(TypePropertyFlags.FallbackProperty) == false)
```

Comment:
There's quite a lot going on on this line - what about instead:
```csharp
if (targetType is ObjectType objType &&
    !objType.AdditionalPropertiesFlags.HasFlag(TypePropertyFlags.FallbackProperty))
```

Reply:
That is much more readable; thanks!